urlcb91e63e47_alexGalushka_E97_Assignment3_tgz-pJ8-cscie97_asn3_test_TestDriverJ8/src/cscie97/asn1/knowledge/engine/Importer.java

```diff
@@ -0,0 +1,157 @@
+	 * @return List<Triple>
+	 */
+	public void collectSquareDeskInfoForSearch ()
+	{
+		try
+		{
+	        Profile tempProvider;
+	        String provId;
+	        String officeId;
+	        String provAndOfficeId;
+	        List<OfficeSpace> officeList;
+	        
+	    	// subject
+	    	Node subjId;
+	    	
+	        // feature
+	    	Node objFeature;
+	    	Node objTraslatedCommonAccessFeat;
+	    	Triple resultingFeatTriple;
+	    	// location
+	    	Node objLocation;
+	    	Triple resultingLocTriple;
+	    	// facility type, and category
+	    	Node objFacility;
+	    	Triple resultingFacTriple;
+	    	// minimum average rating
+	    	Node objRating;
+	    	Triple resultingRatTriple; 
+	        Features tempFeatures;
+	        
+	 
+			for ( Profile provUser: provUserList )
+			{
+				tempProvider = provUser;//.getProfile( "provider" );
+				provId = tempProvider.getGuid().trim().toLowerCase();
+				officeList = tempProvider.getOfficeSpacesList();
+                String[] tempFacTypeCat = {"",""};
+                //String[] tempDates = {"",""};
+				for ( OfficeSpace office:officeList )
+				{
+					officeId = office.getOfficeSpaceGuid().trim().toLowerCase();
+					provAndOfficeId = provId+"&"+officeId;	
+					subjId = new Node(provAndOfficeId);
+					
+					objLocation = new Node(office.getLocation().getSearchableLocation().trim().toLowerCase());
+					resultingLocTriple = new Triple( subjId, locationPredicate, objLocation );
+					resultTripleList.add(resultingLocTriple); //add *
+					
+					tempFeatures = office.getFeatures();
+					
+					for (String feature:tempFeatures.getAllFeatures())
+					{
+						objFeature  = new Node ( feature.trim().toLowerCase() );
+						resultingFeatTriple = new Triple( subjId, featurePredicate, objFeature );
+						resultTripleList.add(resultingFeatTriple); //add *
+					}
+					
+					//common access is considered as a Feature as well, translated to the type such "hasAccessTo_..."
+					for (String com:office.getTranslatedCommonAccessList())
+					{
+						objTraslatedCommonAccessFeat = new Node ( com.trim().toLowerCase() );
+						resultingFeatTriple = new Triple( subjId, featurePredicate, objTraslatedCommonAccessFeat );
+						resultTripleList.add(resultingFeatTriple); //add *
+						
+					}
+					
+					objRating = new Node( office.getRoundedAverageRating().toString().trim().toLowerCase() );
+					resultingRatTriple = new Triple( subjId, ratingPredicate, objRating );
+					resultTripleList.add( resultingRatTriple ); //add *
+					
+					tempFacTypeCat = office.getFacility().getTraslatedCategoryAndType();
+					if ( tempFacTypeCat[0] != null )
+					{
+						if ( !tempFacTypeCat[0].equals("") )
+						{
+							objFacility = new Node ( tempFacTypeCat[0].trim().toLowerCase() );
+					    	resultingFacTriple = new Triple( subjId, facilityAndCategoryPredicate, objFacility );
+					    	resultTripleList.add( resultingFacTriple ); //add *
+						}
+					}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 114:122, 123:131.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.